File: backend/Desmokify.Domain/Entities/DailyCheckIn.cs
namespace Desmokify.Domain.Entities;

public class DailyCheckIn
{
    public int Id { get; set; }
    public int UserId { get; set; }
    public DateTime Date { get; set; }
    public int CigarettesSmoked { get; set; }
    public string? Note { get; set; }
    public DateTime CreatedAt { get; set; }
    public User User { get; set; } = null!;
}

File: mobile/src/types/dailyCheckIn.ts
export interface CreateDailyCheckInRequest {
  cigarettesSmoked: number;
  note?: string;
}

export interface DailyCheckIn {
  id: number;
  date: string;
  cigarettesSmoked: number;
  note: string | null;
  createdAt: string;
}

export interface DailyCheckInStatistics {
  daysSinceQuit: number;
  smokeFreeDays: number;
  currentStreak: number;
  longestStreak: number;
  totalCheckIns: number;
  cigarettesAvoided: number;
  moneySaved: number;
  averageCigarettesPerDay: number;
}